app/src/main/java/com/mk/tasky/agenda/data/AgendaRepositoryImpl.kt

```diff
@@ -203,4 +217,37 @@ class AgendaRepositoryImpl(
             dao.insertEvent(event.toEntity())
         }
     }
+
+    override suspend fun insertEvent(event: AgendaItem.Event, isEdit: Boolean) {
+        // Edge Cases:
+        // - If we get event remotely with different attendees, we should remove all from db and add the new ones, since we could have outdated users
+        insertEventLocally(event)
+        saveEventRemotely(event, isEdit).onFailure {
+            // TODO: Save id on db to later sync with server
+            println(it.message)
+        }
+    }
+
+    private suspend fun saveEventRemotely(event: AgendaItem.Event, isEdit: Boolean) = resultOf {
+        val moshi = Moshi.Builder().build()
+        val jsonAdapter = moshi.adapter(EventDto::class.java)
+        val json: String = jsonAdapter.toJson(event.toDto())
+
+        val photos = event.photos.mapIndexed { index, photo ->
+            val uri = Uri.parse(photo.location)
+            val extension = photoExtensionParser.extensionFromUri(uri)
+            val uriBytes = photoByteConverter.uriToBytes(uri)
+
+            MultipartBody.Part.createFormData(
+                "photo$index",
+                filename = UUID.randomUUID().toString() + "." + extension,
+                uriBytes.toRequestBody()
+            )
+        }
+        // TODO: Add isEdit
+        api.createEvent(
+            body = MultipartBody.Part.createFormData("create_event_request", json),
+            files = photos
+        )
```

Comment:
Uploading potentially a big number of files can take a moment, especially if the connection sucks. A way to make this more convenient for the user is to use WorkManager for this request, so that even if the user goes back or minimizes/closes the app, the photos will be properly uploaded. You can't count on the user staying on the screen till 10 photos are uploaded. 

Also, that way you can easier retry this in case the connection is weak.